nailgun/entities.py

```diff
@@ -6915,3 +6915,95 @@ def update(self, fields=None):
         """
         self.update_json(fields)
         return self.read()
+
+
+class VirtWhoConfig(
+        Entity,
+        EntityCreateMixin,
+        EntityDeleteMixin,
+        EntityReadMixin,
+        EntitySearchMixin,
+        EntityUpdateMixin):
+    """A representation of a VirtWho Config entity."""
+
+    def __init__(self, server_config=None, **kwargs):
+        self._fields = {
+            'blacklist': entity_fields.StringField(),
+            'debug': entity_fields.BooleanField(),
+            'filtering_mode': entity_fields.IntegerField(
+                choices=[0, 1, 2], default=0, required=True),
+            'hypervisor_id': entity_fields.StringField(
+                choices=['hostname', 'uuid', 'hwuuid'],
+                default='hostname', required=True),
+            'hypervisor_password': entity_fields.StringField(),
+            'hypervisor_server': entity_fields.StringField(required=True),
+            'hypervisor_type': entity_fields.StringField(
+                choices=['esx', 'rhevm', 'hyperv', 'xen', 'libvirt'],
+                default='libvirt', required=True),
+            'hypervisor_username': entity_fields.StringField(required=True),
+            'interval': entity_fields.IntegerField(
+                choices=[60, 120, 240, 480, 720], default=120, required=True),
+            'name': entity_fields.StringField(required=True),
+            'no_proxy': entity_fields.StringField(),
+            'organization_id': entity_fields.IntegerField(required=True),
+            'proxy': entity_fields.StringField(),
+            'satellite_url': entity_fields.StringField(required=True),
```

Comment:
Would `URLField` be applicable here?

Reply:
No, the name is misleading. Despite it being `satellite_url` it is actually expecting a hostname (eg `satellite.example.com`). I filed a BZ on this, but it's unlikely it will get renamed.